assembl/graphql/schema.py

```diff
@@ -176,7 +287,43 @@ class Idea(SecureObjectType, SQLAlchemyObjectType):
     class Meta:
         model = models.Idea
         interfaces = (Node, )
-        only_fields = ('id', )
+        only_fields = ('id', 'short_title', )
+
+    posts = SQLAlchemyConnectionField(PostConnection)
+
+    @classmethod
+    def is_type_of(cls, root, context, info):
+        # This is the method defined in SQLAlchemyObjectType where
+        # we changed the isinstance by a type comparison.
+        # For a node query, graphql in
+        # graphene/types/typemap.py(43)resolve_type()
+        # which calls graphql/execution/executor.py(351)get_default_resolve_type_fn()
+        # will try to know the object type from the SA object.
+        # It actually iterate over all registered object types and return
+        # the first one where is_type_of return True.
+        # And here we have in the following order Idea, Question, Thematic.
+        # So a node query on a Thematic or Question was returning the Idea object type.
+        # Here we fix the issue by overriding the is_type_of method
+        # for the Idea type to do a type comparison so that
+        # models.Question/models.Thematic which
+        # inherits from models.Idea doesn't return true
+        if isinstance(root, cls):
+            return True
+        if not is_mapped(type(root)):
+            raise Exception((
+                'Received incompatible instance "{}".'
+            ).format(root))
+        # return isinstance(root, cls._meta.model)  # this was the original code
+        return type(root) == type(cls._meta.model)
```

Comment:
I thought my explanation in the comment was clear. In graphql/execution/executor.py(351)get_default_resolve_type_fn() it iterates over the object types and calls is_type_of of each of them, it returns the first returning true. I don't know how to explain it better.